cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+  })
+
+  describe('adding a filter that excludes all data', () => {
+    before(() => {
+      // Go to the Dashboards list page
+      cy.visit('app/dashboards/list')
+
+      // Click the "Create dashboard" button
+      cy.get('[data-test-subj="newItemButton"]', { timeout: 20000 }).should('be.visible').click()
+      cy.get('[data-test-subj="emptyDashboardWidget"]').should('be.visible')
+      // Change the time to be between Jan 1 2018 and Apr 13, 2018
+      cy.setDashboardDataRange('Apr 13, 2018 @ 00:00:00.000', 'Jan 1, 2018 @ 00:00:00.000')
+
+      // Add all "Filter Bytes Test" visualizations
+      cy.addDashboardPanels('Filter Bytes Test', 'visualization', true)
+
+      // Click the "Add" button and add all "Saved Searches"
+      cy.addDashboardPanels('Filter Bytes Test', 'search', false)
+
+      // Add filter
+      cy.addDashboardFilter('bytes', 'is', '12345678')
+
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').should('be.visible')
+    })
+    it('filters on pie charts', () => {
+      // Check that none of the pie charts are occupied with data (show "No results found")
+      cy.checkElementDoesNotExist('svg > g > g.arcs > path.slice')
+    })
+
+    it('area, bar and heatmap charts filtered', () => {
+      // Check that none of the charts are filled with data
+      cy.checkElementDoesNotExist('svg > g > g.series')
+    })
+
+    it('data tables are filtered', () => {
+      // Check that none of the data tables are filled with data
+      cy.checkElementDoesNotExist('[data-test-subj="paginated-table-body"] [data-cell-content]')
+    })
+
+    it('goal and guages are filtered', () => {
+      // Goal label should be 0, gauge label should be 0%
+      cy.checkElementContainsValue('svg > g > g > text.chart-label', 2, /^0%?$/)
+    })
+
+    it('tsvb time series shows no data message', () => {
+      // The no data message should be visible
+      cy.checkElementExists('[data-test-subj="noTSVBDataMessage"]', 1)
+    })
+
+    it('metric value shows no data', () => {
+      // The metrics should show '-'
+      cy.checkElementContainsValue('.mtrVis__value', 1, /^ - $/)
+    })
+
+    it('tag cloud values are filtered', () => {
+      cy.checkElementComponentDoesNotExist('[data-test-subj="tagCloudVisualization"]', 'svg > g > text')
+    })
+
+    it('tsvb metric is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbMetricValue"]', 1, /^0 custom template$/)
+    })
+
+    it('tsvb top n is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbTopNValue"]', 2, /^0$/)
+    })
+
+    it('saved search is filtered', () => {
+      cy.checkElementDoesNotExist('[data-test-subj="docTableExpandToggleColumn"]')
+    })
+
+    // TODO: Uncomment once https://github.com/elastic/kibana/issues/22561 is fixed
+    // it('timeline is filtered', async () => {
+    //   await dashboardExpect.timelineLegendCount(0)
+    // })
```

Comment:
is it a flaky test, do we need it?

Reply:
After experimenting with the test in Cypress, I found that the timeline visualization does not properly update when applying a filter. For example, after adding a filter that filtered out all data, the timeline continued to be populated with the previous information without any changes. Since this test and all tests involving the timeline visualization expected the timeline to have updated (e.g. by checking whether none of the legend labels existed), they will always fail, and I will remove them from the test suite.